synthesis.go

```diff
@@ -0,0 +1,262 @@
+	for suggestion := range suggestions {
+		suggestionsList = append(suggestionsList, suggestion)
+	}
+	return suggestionsList
+}
+
+// FixCds fixes a CDS given the CDS sequence, a codon table, and a list of functions to solve for.
+func FixCds(sqlitePath string, sequence string, codontable CodonTable, problematicSequenceFuncs []func(string, chan DnaSuggestion, *sync.WaitGroup)) (string, error) {
+	var db *sqlx.DB
+	db = sqlx.MustConnect("sqlite3", sqlitePath)
+	createMemoryDbSql := `
+	CREATE TABLE codon (
+		codon TEXT PRIMARY KEY,
+		aa TEXT
+	);
+
+	CREATE TABLE seq (
+		pos INT PRIMARY KEY
+	);
+
+	CREATE TABLE history (
+		pos INTEGER REFERENCES seq(pos),
+		codon TEXT NOT NULL REFERENCES codon(codon),
+		step INT
+	);
+
+	-- Weights are set on a per position basis for codon harmonization at a later point
+	CREATE TABLE weights (
+		pos INTEGER REFERENCES seq(pos),
+		codon TEXT NOT NULL REFERENCES codon(codon),
+		weight INTEGER
+	);
+
+	CREATE TABLE codonbias (
+		fromcodon TEXT REFERENCES codon(codon),
+		tocodon TEXT REFERENCES codon(codon),
+		gcbias TEXT
+	);
+
+	CREATE TABLE suggestedfix (
+		id INTEGER PRIMARY KEY AUTOINCREMENT,
+		step INTEGER,
+		start INTEGER REFERENCES seq(pos),
+		end INTEGER REFERENCES seq(pos),
+		gcbias TEXT,
+		quantityfixes INTEGER,
+		suggestiontype TEXT
+	);
+`
+	db.MustExec(createMemoryDbSql)
+	// Insert codons
+	weightTable := make(map[string]int)
+	codonInsert := `INSERT INTO codon(codon, aa) VALUES (?, ?)`
+	for _, aminoAcid := range codontable.AminoAcids {
+		for _, codon := range aminoAcid.Codons {
+			db.MustExec(codonInsert, codon.Triplet, aminoAcid.Letter)
+			weightTable[codon.Triplet] = codon.Weight
+
+			codonBias := strings.Count(codon.Triplet, "G") + strings.Count(codon.Triplet, "C")
+			for _, toCodon := range aminoAcid.Codons {
+				if codon.Triplet != toCodon.Triplet {
+					toCodonBias := strings.Count(toCodon.Triplet, "G") + strings.Count(toCodon.Triplet, "C")
+					switch {
+					case codonBias == toCodonBias:
+						db.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "NA")
+					case codonBias > toCodonBias:
+						db.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "AT")
+					case codonBias < toCodonBias:
+						db.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "GC")
+					}
+				}
+			}
+		}
+	}
+
+	// Insert seq and history
+	pos := 0
+	for i := 0; i < len(sequence)-3; i = i + 3 {
+		codon := sequence[i : i+3]
+		db.MustExec(`INSERT INTO seq(pos) VALUES (?)`, pos)
+		db.MustExec(`INSERT INTO history(pos, codon, step) VALUES (?, ?, 0)`, pos, codon)
+		db.MustExec(`INSERT INTO weights(pos, codon, weight) VALUES (?,?,?)`, pos, codon, weightTable[codon])
+		pos++
+	}
+
+	var err error
+	// For a maximum of 100 iterations, see if we can do better
+	for i := 1; i < 100; i++ {
```

Comment:
```suggestion
	for i := 1; i < ITERATION; i++ {
```
this doesn't seem like it should be hardcode.